// @azure/arm-appcontainers/dist-esm/src/operations/containerAppsAuthConfigs.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing ContainerAppsAuthConfigs operations. */
export class ContainerAppsAuthConfigsImpl {
    /**
     * Initialize a new instance of the class ContainerAppsAuthConfigs class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get the Container App AuthConfigs in a given resource group.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param containerAppName Name of the Container App.
     * @param options The options parameters.
     */
    listByContainerApp(resourceGroupName, containerAppName, options) {
        const iter = this.listByContainerAppPagingAll(resourceGroupName, containerAppName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: () => {
                return this.listByContainerAppPagingPage(resourceGroupName, containerAppName, options);
            }
        };
    }
    listByContainerAppPagingPage(resourceGroupName, containerAppName, options) {
        return __asyncGenerator(this, arguments, function* listByContainerAppPagingPage_1() {
            let result = yield __await(this._listByContainerApp(resourceGroupName, containerAppName, options));
            yield yield __await(result.value || []);
            let continuationToken = result.nextLink;
            while (continuationToken) {
                result = yield __await(this._listByContainerAppNext(resourceGroupName, containerAppName, continuationToken, options));
                continuationToken = result.nextLink;
                yield yield __await(result.value || []);
            }
        });
    }
    listByContainerAppPagingAll(resourceGroupName, containerAppName, options) {
        return __asyncGenerator(this, arguments, function* listByContainerAppPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByContainerAppPagingPage(resourceGroupName, containerAppName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Get the Container App AuthConfigs in a given resource group.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param containerAppName Name of the Container App.
     * @param options The options parameters.
     */
    _listByContainerApp(resourceGroupName, containerAppName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, containerAppName, options }, listByContainerAppOperationSpec);
    }
    /**
     * Get a AuthConfig of a Container App.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param containerAppName Name of the Container App.
     * @param authConfigName Name of the Container App AuthConfig.
     * @param options The options parameters.
     */
    get(resourceGroupName, containerAppName, authConfigName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, containerAppName, authConfigName, options }, getOperationSpec);
    }
    /**
     * Create or update the AuthConfig for a Container App.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param containerAppName Name of the Container App.
     * @param authConfigName Name of the Container App AuthConfig.
     * @param authConfigEnvelope Properties used to create a Container App AuthConfig
     * @param options The options parameters.
     */
    createOrUpdate(resourceGroupName, containerAppName, authConfigName, authConfigEnvelope, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            containerAppName,
            authConfigName,
            authConfigEnvelope,
            options
        }, createOrUpdateOperationSpec);
    }
    /**
     * Delete a Container App AuthConfig.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param containerAppName Name of the Container App.
     * @param authConfigName Name of the Container App AuthConfig.
     * @param options The options parameters.
     */
    delete(resourceGroupName, containerAppName, authConfigName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, containerAppName, authConfigName, options }, deleteOperationSpec);
    }
    /**
     * ListByContainerAppNext
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param containerAppName Name of the Container App.
     * @param nextLink The nextLink from the previous successful call to the ListByContainerApp method.
     * @param options The options parameters.
     */
    _listByContainerAppNext(resourceGroupName, containerAppName, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, containerAppName, nextLink, options }, listByContainerAppNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByContainerAppOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/containerApps/{containerAppName}/authConfigs",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AuthConfigCollection
        },
        default: {
            bodyMapper: Mappers.DefaultErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.containerAppName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/containerApps/{containerAppName}/authConfigs/{authConfigName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AuthConfig
        },
        default: {
            bodyMapper: Mappers.DefaultErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.containerAppName,
        Parameters.authConfigName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/containerApps/{containerAppName}/authConfigs/{authConfigName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.AuthConfig
        },
        default: {
            bodyMapper: Mappers.DefaultErrorResponse
        }
    },
    requestBody: Parameters.authConfigEnvelope,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.containerAppName,
        Parameters.authConfigName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/containerApps/{containerAppName}/authConfigs/{authConfigName}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.DefaultErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.containerAppName,
        Parameters.authConfigName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByContainerAppNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AuthConfigCollection
        },
        default: {
            bodyMapper: Mappers.DefaultErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.containerAppName,
        Parameters.nextLink
    ],
    headerParameters: [Parameters.accept],
    serializer
};
